lib/vrl/parser/src/parser.lalrpop

```diff
@@ -0,0 +1,511 @@
+        let mut alternative = alternative;
+
+        for Node { span, mut node } in alternatives {
+            node.alternative = alternative;
+            let node = Node::new(span, Expr::IfStatement(Node::new(span, node)));
+
+            alternative = Some(Node::new(node.span(), Block(vec![node])));
+        }
+
+        IfStatement { predicate, consequent, alternative }
+};
+
+ElseIf: IfStatement = "else" "if" <predicate: Sp<Predicate>> <consequent: Sp<Block>> => {
+    IfStatement { predicate, consequent, alternative: None }
+};
+
+Predicate: Predicate = {
+    Box<ArithmeticExpr> => Predicate::One(<>),
+    "(" <v:(<AssignmentExpr> ";")+> <e:(<AssignmentExpr>)?> ")" => {
+        let expressions = match e {
+            None => v,
+            Some(e) => {
+                let mut v = v;
+                v.push(e);
+                v
+            }
+        };
+
+        Predicate::Many(expressions)
+    },
+};
+
+// -----------------------------------------------------------------------------
+// containers
+// -----------------------------------------------------------------------------
+
+Container: Container = {
+    Box<Sp<Group>> => Container::Group(<>),
+    Sp<Block> => Container::Block(<>),
+    Sp<Array> => Container::Array(<>),
+    Sp<Object> => Container::Object(<>),
+};
+
+Group: Group = "(" <AssignmentExpr> ")" => Group(<>);
+
+Block: Block = "{" NonterminalNewline* <Exprs>  "}" => Block(<>);
+
+Array: Array = {
+    "[" NonterminalNewline* "]" => Array(vec![]),
+    "[" NonterminalNewline* <CommaMultiline<ArithmeticExpr>> "]" => Array(<>),
+};
+
+Object: Object = {
+    "{" NonterminalNewline* "}" => Object(BTreeMap::default()),
+    "{" NonterminalNewline* <CommaMultiline<(<Sp<String>> ":" <ArithmeticExpr>)>> "}" => {
+        let object =<>.into_iter().collect::<BTreeMap<_, _>>();
+        Object(object)
+    },
+};
+
+// -----------------------------------------------------------------------------
+// literals
+// -----------------------------------------------------------------------------
+
+Literal: Literal = {
+    String => Literal::String(<>),
+    Integer => Literal::Integer(<>),
+    Float => Literal::Float(<>),
+    Boolean => Literal::Boolean(<>),
+    Null => Literal::Null,
+    Regex => Literal::Regex(<>),
+    Timestamp => Literal::Timestamp(<>),
+};
+
+String: String = "string literal" => <>.unescape();
+Integer: i64 = "integer literal";
+Float: NotNan<f64> = "float literal";
+Boolean: bool = { "true" => true, "false" => false };
+Null: () = "null";
+Regex: String = "regex literal" => <>.replace("\\'", "'");
+Timestamp: String = "timestamp literal" => <>.replace("\\'", "'");
+
+// -----------------------------------------------------------------------------
+// macros
+// -----------------------------------------------------------------------------
+
+#[inline]
+Box<T>: Box<T> = T => Box::new(<>);
+
+#[inline]
+Sp<T>: Node<T> = <l: @L> <rule: T> <r: @R> => Node::new(span(l, r), rule);
```

Comment:
Does `Sp` stand for `Span`?

Reply:
Yes

Reply:
Would there be a name clash if this was called `Span`? Otherwise suggesting to change it.